### PR TITLE
Points in QC Charts are not displayed in accordance with results capture date

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

The javascript that renders the chart expects the following format

https://github.com/senaite/senaite.core/blob/master/bika/lims/browser/js/bika.lims.graphics.controlchart.js#L180-L181

which does not match with what `self.ulocalized_time(results_date, long_format=1)` returns
https://github.com/senaite/senaite.core/blob/master/bika/lims/browser/analyses.py#L454-L455

Also, this `ulocalized_time` does not differentiate between PM and AM, so needs to be reviewed too.

## Current behavior before PR

qc_chart_old

## Desired behavior after PR is merged

qc_chart_after

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bika/lims/browser/instrument.py [hunk 1]
```diff
@@ -532,9 +532,10 @@ def folderitems(self):
                     error_amount = ((target / 100) * error) if target > 0 else 0
                     upper = smax + error_amount
                     lower = smin - error_amount
-
+                    cap_date = obj.getResultCaptureDate()
+                    cap_date = cap_date and cap_date.strftime('%Y-%m-%d %I:%M %p') or ''
```
Comment (ramonski) [inline bika/lims/browser/instrument.py]: maybe better use `isinstance(cap_date, DateTime)` or add a `is_date` function to the API. I've seen already cases where we got weird stuff somehow...

Comment (xispa) [reply bika/lims/browser/instrument.py]: Yes, you are right, will add this change. Thanks!

File: bika/lims/api.py [hunk 2]
```diff
@@ -1166,3 +1170,33 @@ def is_uid(uid, validate=False):
...
+    :type date: DateTime
+    :return: True if a valid date
+    :rtype: bool
+    """
+    if not date:
+        return False
+    return isinstance(date, DateTime) or isinstance(date, datetime)
+
+
+def get_date(value):
```
Comment (ramonski) [inline bika/lims/api.py]: Would you mind to change the name to `to_date` and pass as well a `default` value?

File: bika/lims/api.py [hunk 3]
```diff
@@ -1166,3 +1170,33 @@ def is_uid(uid, validate=False):
...
+    """Checks if the passed in value is a valid Zope's DateTime
+
+    :param date: The date to check
+    :type date: DateTime
+    :return: True if a valid date
+    :rtype: bool
+    """
+    if not date:
+        return False
+    return isinstance(date, DateTime) or isinstance(date, datetime)
```
Comment (ramonski) [inline bika/lims/api.py]: change to `return isinstance(date, (DateTime, datetime))`

**Comment by lemoene** [comment]
We saw, thanks. Am adding a date range at the moment

**Pull request merged**